system/common.xml

```diff
@@ -49,17 +49,49 @@
 	</target>
 
 	<target name="clone_stf">
-		<echo message="Cloning from: ${stf_repo} -b ${stf_branch}"/>
+		<echo message="git clone -q ${stf_repo} "/>
 		<exec executable="git" failonerror="false" dir="${SYSTEMTEST_ROOT}">
-			<arg value="clone" />
-			<arg value="--depth" />
-			<arg value="1" />
-			<arg value="--single-branch" />
-			<arg value="-b"/>
-			<arg value="${stf_branch}"/>
+			<arg value="clone" />		
+			<arg value="-q"/>
 			<arg value="${stf_repo}"/>
-			<arg value="stf"/>
 		</exec>
+
+		<echo message="git rev-parse ${stf_branch} "/>
+		<exec executable="git" failonerror="false" dir="${SYSTEMTEST_ROOT}/stf" outputproperty="stf_sha" resultproperty="code">
+			<arg value="rev-parse"/>
+			<arg value="${stf_branch}"/>
+		</exec>
+		<if>
+ 			<equals arg1="${code}" arg2="128"/>			
+			<then>
+				<echo message="git rev-parse origin/${stf_branch}"/>
+				<exec executable="git" failonerror="false" dir="${SYSTEMTEST_ROOT}/stf" outputproperty="stf_sha" resultproperty="code2">
+					<arg value="rev-parse"/>
+					<arg value="origin/${stf_branch}"/>
+				</exec>
+			</then>
+		</if>
+		
+		<if>
+ 			<equals arg1="${code2}" arg2="128"/>
+			<then>
+				<echo message="This branch does not exist!"/>
+			</then>
+		</if>
```

Comment:
Could we move the if block inside the above if (after line 71)? We only need to check code2 if we execute line 68.

Reply:
changed it pl review